SU-HSDO/suhumsci
# [STN-60] Multi-theme Sass build

# Summary

Configures a Sass build in the Humsci Basic theme that generates different stylesheets for each of the sub-themes.

- Uses theme specific Grunt task to run Sass and PostCSS tasks on Sass files in the theme
- NPM scripts are setup to run the Grunt tasks (`npm run watch` and `npm run build:sass`)
- The build generates a stylesheet for each of the 3 themes in the `dist/css` folder
- The 3 Drupal themes are configured to load their respective stylesheets found in Humsci Basic
- linting is also set up as npm script (`npm run lint:sass`) and will be run during the watch task at each change
- sass-true tests have been set up with mocha (`npm run test:sass`)

# TODO

-  ✅ We should setup a Grunt task for linting the Sass stylesheets: https://stylelint.io/
- ✅ Setup mixin that would allow adding styles for multiple themes
- ✅ play around with a card to do Adam's version of set up with sass themes in subthemes https://github.com/SU-HSDO/suhumsci/pull/438
- ✅ set up true Sass tests

# Need Review By (Date)
_['10/30', 'asap', etc.]_

# Urgency
_['low', 'medium', 'high', etc.]_

# Steps to Test
Make some changes to the sass files in humsci_basic and see if these scripts work with them:

- `npm run build:css` should build your new sass to css and apply postCss where necessary
- `npm run lint:sass` should check your new sass for errors (add an error if you'd like)
- `npm run watch` will start watching the humsci_basic src/scss folder. Make a change and see it run.
- `npm run test:sass` make sure sass tests pass

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
